src/main/webapp/3Dscript.js

```diff
@@ -208,3 +244,74 @@ function animate() {
   renderer.render(scene, camera);
 };
 
+/* What to do on mouse click. */
+function onClick(event) {
+  /* If a cylinder was the previous thing clicked, unscale it. */
+  if (oldIndex!= -1) {
+    direction.getMatrixAt(oldIndex, instanceMatrix );
+    matrix.multiplyMatrices( instanceMatrix, scaleInverseMatrix );
+    direction.setMatrixAt( oldIndex, matrix );
+    direction.instanceMatrix.needsUpdate = true;
+    oldIndex= -1;
+  }
+
+  event.preventDefault();
+  mouse.x = ( event.clientX / window.innerWidth ) * 2 - 1;
+  mouse.y = - ( event.clientY / window.innerHeight ) * 2 + 1;
+
+  /* Get array of objects intersected by raycaster */
+  raycaster.setFromCamera( mouse, camera );
+  const intersects = raycaster.intersectObjects(scene.children, true);
+
+  /* Shows properties of first instance of intersected cylinder. */
+  for (let i=0; i < intersects.length; i++) {
+    if ( intersects[i].object.type == 'Mesh' ) {
```

Comment:
Can you separate out this for loop body into a function. Also you for (intersect of intersection) or some better variable naming convention aha.

Reply:
makes sense. will make better variable name here, using another branch to add additional feature and modularize codes, so i think it would be redundant to do here too.